src/water_cmd.cpp

```diff
@@ -475,11 +475,12 @@ CommandCost CmdBuildCanal(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32
 			return_cmd_error(STR_ERROR_FLAT_LAND_REQUIRED);
 		}
 
+		bool water = IsWaterTile(current_tile);
+
 		/* can't make water of water! */
-		if (IsTileType(current_tile, MP_WATER) && (!IsTileOwner(current_tile, OWNER_WATER) || wc == WATER_CLASS_SEA)) continue;
+		if (water && (!IsTileOwner(current_tile, OWNER_WATER) || wc == WATER_CLASS_SEA)) continue;
```

Comment:
Please completely remove this if-statement. It has no reason to exist anymore.

Reply:
@SamuXarick points out I am wrong, and it should exist. But indeed, it should at least be `IsWaterTile` instead of the `MP_WATER` check :)